scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -2151,7 +2151,7 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
       case Ident(_) => termName()
       case LeftBrace() => dropTrivialBlock(expr(location = NoStat, allowRepeated = true))
       case KwThis() =>
-        val qual = atPos(in.tokenPos, in.prevTokenPos)(Name.Anonymous()); next();
+        val qual = autoPos(Name.Anonymous()); next();
```

Comment:
what does this look like?

Reply:
`q"asdsa $this"`. 

In this case, the previous logic wasn't changed - `Name.Anonymous` doesn't point at `this`.
Would it be ok to change that?

Reply:
yes!